crates/ntx-builder/src/state/account.rs

```diff
@@ -3,9 +3,59 @@ use std::collections::{HashMap, VecDeque};
 use miden_node_proto::domain::{account::NetworkAccountPrefix, note::NetworkNote};
 use miden_objects::{
     account::{Account, AccountDelta, AccountId, delta::AccountUpdateDetails},
+    block::BlockNumber,
     note::Nullifier,
 };
 
+// INFLIGHT NETWORK NOTE
+// ================================================================================================
+
+/// An unconsumed network note that may have failed to execute.
+///
+/// The block numbers at which the network note was attempted are approximate and may not
+/// reflect the exact block number for which the execution attempt failed. The actual block
+/// will likely be soon after the number that is recorded here.
+#[derive(Debug, Clone)]
+pub struct InflightNetworkNote {
+    note: NetworkNote,
+    attempted_at: Vec<BlockNumber>,
```

Comment:
I would either:

1. Replace this with a separate `count` and `last_attempt: Option<BlockNumber>`, or
2. document that this is bounded by the maximum number of failed attempts allowed per note

I would probably go with (1) to keep the reasoning local to the type.